packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/include/camera_avfoundation/FLTCaptureDeviceControlling.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import AVFoundation;
+@import Foundation;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol FLTCaptureDeviceControlling <NSObject>
+
+// Device
+- (NSString *)uniqueID;
+
+// Position/Orientation
+- (AVCaptureDevicePosition)position;
+
+// Format/Configuration
+- (AVCaptureDeviceFormat *)activeFormat;
+- (NSArray<AVCaptureDeviceFormat *> *)formats;
+- (void)setActiveFormat:(AVCaptureDeviceFormat *)format;
+
+// Flash/Torch
+- (BOOL)hasFlash;
+- (BOOL)hasTorch;
+- (BOOL)isTorchAvailable;
+- (AVCaptureTorchMode)torchMode;
+- (void)setTorchMode:(AVCaptureTorchMode)torchMode;
+- (BOOL)isFlashModeSupported:(AVCaptureFlashMode)mode;
+
+// Focus
+- (BOOL)isFocusPointOfInterestSupported;
+- (BOOL)isFocusModeSupported:(AVCaptureFocusMode)mode;
+- (void)setFocusMode:(AVCaptureFocusMode)focusMode;
+- (void)setFocusPointOfInterest:(CGPoint)point;
+
+// Exposure
+- (BOOL)isExposurePointOfInterestSupported;
+- (void)setExposureMode:(AVCaptureExposureMode)exposureMode;
+- (void)setExposurePointOfInterest:(CGPoint)point;
+- (float)minExposureTargetBias;
+- (float)maxExposureTargetBias;
+- (void)setExposureTargetBias:(float)bias completionHandler:(void (^_Nullable)(CMTime))handler;
+- (BOOL)isExposureModeSupported:(AVCaptureExposureMode)mode;
+
+// Zoom
+- (float)maxAvailableVideoZoomFactor;
+- (float)minAvailableVideoZoomFactor;
+- (float)videoZoomFactor;
+- (void)setVideoZoomFactor:(float)factor;
+
+// Camera Properties
+- (float)lensAperture;
+- (CMTime)exposureDuration;
+- (float)ISO;
+
+// Configuration Lock
+- (BOOL)lockForConfiguration:(NSError **)error;
+- (void)unlockForConfiguration;
+
+// Frame Duration
+- (CMTime)activeVideoMinFrameDuration;
+- (void)setActiveVideoMinFrameDuration:(CMTime)duration;
+- (CMTime)activeVideoMaxFrameDuration;
+- (void)setActiveVideoMaxFrameDuration:(CMTime)duration;
+
+@end
+
+@protocol FLTCaptureInput <NSObject>
+@property(nonatomic, readonly) NSArray<AVCaptureInputPort *> *ports;
+@end
+
+@protocol FLTCaptureDeviceInputFactory <NSObject>
+- (nullable id<FLTCaptureInput>)deviceInputWithDevice:(id<FLTCaptureDeviceControlling>)device
+                                                error:(NSError **)error;
+@end
+
+@interface AVCaptureDevice (FLTCaptureDeviceControlling) <FLTCaptureDeviceControlling>
```

Comment:
I'd just call the protocol `FLTCaptureDevice` since it's a noun (rather than a capability), [as discussed in the design doc](https://docs.google.com/document/d/1XsaulkJA6_ZSpM7chkQLhQY25sQqhEMiqHMYzw8H85o/edit?resourcekey=0-_cUjF1c0iBvRLKfV-3gK5A&tab=t.0#heading=h.a6hzk7tvmg6z)

Reply:
Done